Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -59,7 +51,79 @@ public static MapElementLayer parse(@NonNull String kml) throws KMLParseExceptio
     }
   }
 
-  MapElementLayer internalParse(@NonNull String kml, @NonNull MapFactories factory) {
+  @VisibleForTesting
+  @NonNull
+  MapElementLayer internalParse(@NonNull String kml, @NonNull MapFactories factory)
+      throws XmlPullParserException, IOException, KMLParseException {
+    mFactory = factory;
+    mLayer = mFactory.createMapElementLayer();
+    try (InputStream stream = new ByteArrayInputStream(kml.getBytes(UTF_8))) {
+      mParser.setInput(stream, null);
+      mParser.nextTag();
+      mNameSpace = mParser.getNamespace();
+      parsePlacemark();
+    }
+    return mLayer;
+  }
+
+  private void parsePlacemark() throws IOException, XmlPullParserException, KMLParseException {
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("Placemark")) {
```

Comment:
Are names other than Placemark ok to ignore here?

Reply:
Yes: Placemarks are the only element that hold a geometry shape.